### ISSUE TITLE
nothing is happening

### ISSUE THREAD

**Issue by itsroki** [hackerspace-adelaide/XL320#7]
does anyone know what do we have to change in code to make it work because i can not make my motor run?

**Comment by sighmon** [comment]
@itsroki I'll see if I can find the ones I was playing with this weekend and boot them up again.

**Comment by itsroki** [comment]
thanks, that would be great!

**Comment by itsroki** [comment]
i have figured it out (how to make motor run), but i don't know how to give him position to go to because every time i run the servo_example, dynamixel just runs in circles. any suggestions?

**Comment by benedikt966** [comment]
I dont have the motors anymore, but for me it worked with the library at the "bene" branch very well - together with the xl320 library

**Comment by benedikt966** [comment]
https://1drv.ms/f/s!Ankn9OnwjyR2geFR-GZQqe1xCRvmbg - here are the two libraries and one example code, that worked with my teensy board and arduino board. But I dont have these here anymore to test it. You have to look, the motors have to modes: joint mode and wheel mode. So if you are in joint mode, you can give the motor positions and if you are in wheel mode, you can give the motor driving speeds.

**Comment by benedikt966** [comment]
And be careful, 
-if you work with the ROBOTIS software directly. It happens very fast, that you crash the software at the XL320 Motors and have big problems to reset them again. 
To give the motor Positions - normally use setPosition() ore setJointPosition or sth like that. Thats all I want to give you at your way with these motors =)
But all in one these are nice motors, which can do a lot, if you use them rigth and dont have to high expecations for the accuracy (<5% deviations)

**Comment by itsroki** [comment]
thank you very much, i will look at it now and tomorrow. i'm sure i will make it work, you have been very helpful!

**Comment by itsroki** [comment]
i have managed to fix the problem, i only had to change operating mode to JOINT in roboplus2 software to make it move to wanted position, thanks!

**Comment by ryantjhi** [comment]
> i have figured it out (how to make motor run), but i don't know how to give him position to go to because every time i run the servo_example, dynamixel just runs in circles. any suggestions?
 i have the same problem how you figured it out ? (only red led out from motor)

**Comment by itsroki** [comment]
> i have figured it out (how to make motor run), but i don't know how to give him position to go to because every time i run the servo_example, dynamixel just runs in circles. any suggestions?
> i have the same problem how you figured it out ? (only red led out from motor)

You need to set the motor parameters in robotis roboplus software, important is to set operating mode to JOINT if you want to give him position, or WHEEL if you want to give him speed.

**Comment by NikeetPandit** [comment]
@itsroki  Hope all is well. What did you do to initially make the motor run? Having issues.

**Comment by NikeetPandit** [comment]
@benedikt966 are you able to re-upload your two libraries? Would be greatly appreciated. The onedrive link seems to be offline unfortunately.

**Comment by itsroki** [comment]
Wow it has been a long time since then, what problems do you have? You have to set everything in robotis software before you try to run.

Dohvati Outlook za iOS<https://aka.ms/o0ukef>
________________________________
Šalje: NikeetPandit <notifications@github.com>
Poslano: Monday, February 1, 2021 10:56:23 PM
Prima: hackerspace-adelaide/XL320 <XL320@noreply.github.com>
Kopija: itsroki <roman.kos1997@hotmail.com>; Mention <mention@noreply.github.com>
Predmet: Re: [hackerspace-adelaide/XL320] nothing is happening (#7)


@benedikt966<https://github.com/benedikt966> are you able to re-upload your two libraries? Would be greatly appreciated. The onedrive link seems to be offline unfortunately.

**Comment by NikeetPandit** [comment]
@itsroki  Wow! Thank you for your quick reply! I think I am out of my scope of knowledge. But what is meant by setting everything in the robotis software before trying to run? 
Is it incorrect to assume that this standalone library and examples can be downloaded on Teensy to control the XL320 servos as a simple starting point?

**Comment by itsroki** [comment]
Yes you can download examples but first you need to set parameters in robotis software, google that because i don remember. Example. In robotis software you need to set your motor in joint mode if you want to give him final position to go to. You do that in robotis, otherwise you can not use example code to move your motor.

Dohvati Outlook za iOS<https://aka.ms/o0ukef>
________________________________
Šalje: NikeetPandit <notifications@github.com>
Poslano: Monday, February 1, 2021 11:16:40 PM
Prima: hackerspace-adelaide/XL320 <XL320@noreply.github.com>
Kopija: itsroki <roman.kos1997@hotmail.com>; Mention <mention@noreply.github.com>
Predmet: Re: [hackerspace-adelaide/XL320] nothing is happening (#7)


@itsroki<https://github.com/itsroki> Wow! Thank you for your quick reply! I think I am out of my scope of knowledge. But what is meant by setting everything in the robotis software before trying to run?
Is it incorrect to assume that this standalone library and examples can be downloaded on Teensy to control the XL320 servos as a simple starting point?

**Comment by NikeetPandit** [comment]
Okay, thank you. I will try and let you know if anything gets working! Thanks

**Comment by NikeetPandit** [comment]
Followed this method by Walker's blog: http://walker.gosrich.com/posts/xl320.html ...
Simply connected as he wired and downloaded the software onto Arduino - worked (must initially have default baud to 1Mbs). For teensy - had to change the int_16 variables to unsigned short and worked no problem. If you study the datasheet for XL-320 servo and cross reference to his example (and have some exp. with hex) you should have no problem sending any command to the servo! Good luck.
